lib/grpc_mock/mocked_call.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'grpc'
+
+module GrpcMock
+  class MockedCall
+    attr_reader :deadline, :metadata
+
+    def initialize(metadata: {})
+      @metadata = MockedCall.sanitize_metadata(metadata)
+      @deadline = Time.now + 5
+    end
+
+    def multi_req_view
+      GRPC::ActiveCall::MultiReqView.new(self)
+    end
+
+    def single_req_view
+      GRPC::ActiveCall::SingleReqView.new(self)
+    end
+
+    def self.sanitize_metadata(metadata)
+      raise TypeError, "got <#{metadata.class}>, want <Hash>" unless metadata.is_a?(Hash)
+
+      headers = []
+      metadata.each do |key, value|
+        raise TypeError, "bad type for key parameter" unless key.is_a?(String) || key.is_a?(Symbol)
+
+        key = key.to_s
+        raise ArgumentError, "'#{key}' is an invalid header key" unless key.match?(/\A[a-z0-9-_.]+\z/i)
+        raise ArgumentError, "Header values must be of type string or array" unless value.is_a?(String) || value.is_a?(Array)
+
+        Array(value).each do |elem|
+          raise TypeError, "Header value must be of type string" unless elem.is_a?(String)
+
+          unless key.end_with?('-bin')
+            raise ArgumentError, "Header value '#{elem}' has invalid characters" unless elem.match(/\A[ -~]+\z/)
+
+            elem = elem.strip
```

Comment:
gRPC specifies that trailing/leading whitespaces **may** be trimmed. I guess it depends on underlying HTTP implementations.

Another option would be to error here too, do you have any ideas?

Reply:
> https://github.com/grpc/grpc/blob/c483ece5dd789d55e95c34a89b02acd6eee720ac/doc/PROTOCOL-HTTP2.md

> ASCII-Value should not have leading or trailing whitespace. If it contains leading or trailing whitespace, it may be stripped.

ah, I missed that. No problem, then.